File: app/src/main/java/com/tobiasfried/rasa/constants/TeaType.java
package com.tobiasfried.rasa.constants;

public enum TeaType {

    WHITE(0),
    GREEN(1),
    OOLONG(2),
    BLACK(3),
    PUERH(4),
    HERBAL(5),
    OTHER(6);

    private final int code;

    TeaType(int code) {
        this.code = code;
    }

    public int getCode() {
        return this.code;
    }

    public static TeaType get(int i) {
        for (TeaType type : TeaType.values()) {
            if (type.code == i) return type;
        }
        throw new IllegalArgumentException("That tea type does not exist.");
    }

}

File: app/src/main/java/com/tobiasfried/rasa/constants/TeaType.kt
package com.tobiasfried.rasa.constants

enum class TeaType private constructor(val code: Int) {

    WHITE(0),
    GREEN(1),
    OOLONG(2),
    BLACK(3),
    PUERH(4),
    HERBAL(5),
    OTHER(6);


    companion object {

        operator fun get(i: Int): TeaType {
            for (type in TeaType.values()) {
                if (type.code == i) return type
            }
            throw IllegalArgumentException("That tea type does not exist.")
        }
    }

}
